File: arrays/two_sum.cpp
class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> seen;

    for (int i = 0; i < nums.size(); i++) {
      int other = target - nums[i];

      if (seen.find(other) != seen.end()) {
        return {seen[other], i};
      } else {
        seen[nums[i]] = i;
      }
    }

    return {};
  }
};

File: arrays/two_sum.go
func twoSum(nums []int, target int) []int {
	seen := make(map[int]int)

	for idx, element := range nums {
		leftover := target - element
		if _, ok := seen[leftover]; ok {
			return []int{seen[leftover], idx}
		}
		seen[element] = idx
	}

	return nil
}
